packages/ui/src/helpers/typeGuards.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Checks if `value` is an Object (excludes arrays and functions)
+ *
+ * @param value
+ * @returns
+ */
+export const isObject = (value: unknown): value is object =>
+  value != null && !Array.isArray(value) && typeof value === 'object';
+
+/**
+ * Checks if `value` is a string
+ *
+ * @param value
+ * @returns
+ */
+export const isString = (value: unknown): value is string =>
+  typeof value === 'string';
+
+/**
+ * Checks if `value` is undefined
+ *
+ * @param value
+ * @returns
+ */
+export const isUndefined = <T>(value: T | undefined): value is T =>
```

Comment:
Maybe I am misunderstanding the usage of the generic, but it seems to me we would want the return type of this type predicate to return `value is undefined`?

Reply:
Maybe we can use `unknown` instead? Not sure if the generic type is necessary

Reply:
Missed on my first review but after seeing @zchenwei callout regarding the generic, not sure how well TS supports using generics with arrow functions and once we add stricter linting on for the UI package we will most likely see errors here (also breaks prettier i guess?):
<img width="1397" alt="image" src="https://user-images.githubusercontent.com/15002885/213268437-77fd725e-4ead-43c3-b421-46ddd4e79707.png">

Reply:
Good call, no need for generics, updated!